### PR TITLE
[INFRA-2118] Add function to run JMH benchmarks

### PR DESCRIPTION
**Pull request by AbhyudayaSharma**
Uses the added Maven profile from https://github.com/jenkinsci/plugin-pom/pull/206 and https://github.com/jenkinsci/jenkins-test-harness/pull/135
 and adds a utility function to run benchmarks in Jenkinsfiles.

@oleg-nenashev @runzexia @supun94

### CODE REVIEW

File: vars/runBenchmarks.groovy [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env groovy
+
+/**
+ * Run JMH benchmarks and archives results
+ * @param artifacts archive these JMH reports
+ * @since TODO
+ */
+def call(String artifacts = null) {
+    // TODO: Use Lockable Resources plugin
```
Comment (oleg-nenashev) [inline vars/runBenchmarks.groovy]: I am not sure we can merge it without it. Maybe it makes sense to create an INFRA ticket to @olblak to get it set up. Maybe 2 parallel runs is a good limit for now.

File: vars/runBenchmarks.groovy [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            infra.checkout()
+        }
+
+        stage('Run Benchmarks') {
+            List<String> mvnOptions = ['test', '-Dbenchmark']
+            infra.runMaven(mvnOptions)
+        }
+
+        stage('Archive reports') {
+            if (artifacts != null) {
```
Comment (oleg-nenashev) [inline vars/runBenchmarks.groovy]: Undefined variable

File: vars/runBenchmarks.groovy [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                archiveArtifacts artifacts: artifacts
```
Comment (oleg-nenashev) [inline vars/runBenchmarks.groovy]: Would it also make sense to generate a http://jmh.morethan.io/ viewer link and put it to the console log for now?

File: vars/runBenchmarks.groovy [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env groovy
```
Comment (oleg-nenashev) [inline vars/runBenchmarks.groovy]: There should be also a `runBenchmarks.txt` HTML text file. It is used for generating built-in Documentation in Jenkins IIRC

File: vars/runBenchmarks.groovy (hunk 2 above)
Comment (AbhyudayaSharma) [reply vars/runBenchmarks.groovy]: `artifacts` is a parameter to the function

File: vars/runBenchmarks.groovy (hunk 1 above)
Comment (AbhyudayaSharma) [reply vars/runBenchmarks.groovy]: Created https://issues.jenkins-ci.org/browse/INFRA-2141

Comment (olblak) [reply vars/runBenchmarks.groovy]: the lockable plugin is already installed

Comment (AbhyudayaSharma) [reply vars/runBenchmarks.groovy]: @olblak We need you to create a resource that we can use for throttling benchmarks running on `highmem` nodes.

Comment (olblak) [reply vars/runBenchmarks.groovy]: so something like lock('runBenchmarks') {
}

should be fine

**Comment by AbhyudayaSharma** [comment]
Yes, that would be great!

**Comment by oleg-nenashev** [comment]
" IMHO we can ship it once JTH and Plugin POM patches are out" @olblak They are not :) No worries, we will get there soon

**Pull request merged**